### DISCUSSION TITLE
Simulate a Generic RF433 Module using SX1278

### DISCUSSION THREAD

**Discussion by Jev1337** [jgromes/RadioLib#857] [Q&A]
Hello!
I recently bought a LoRa RA-01, reason was mainly to replace my Generic (the cheap one) RF433 Tx Module (to open my garage door) since its range is very limited (up to 20meters), to my surprise I noticed that the Generic RF433 Tx Module uses ASK modulation that SX1278 does not directly support, but after some research, I found out that OOK modulation could work.

The data I send is as such:

Byte data[] = {0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07} (this is just an example)

I already tried Enable with FSK and enabling OOK modulation in the constructor, but using the Generic RF433 Receiver, I see nothing, just the random bits 0s and 1s caused by noise.

How do I manage to make my SX1278 "compatible" with the generic receiver? Is it impossible to do so? Has anyone here configured the SX1278 in a way a generic 433RF receiver could understand?

One last thing, how do I setup these informations using RadioLib for Tx? (imported from rf433_send library):
```
rfsend_builder(
    RfSendEncoding::TRIBIT,  //Encoding
    PIN_RFOUT,  
    RFSEND_DEFAULT_CONVENTION,  // Do we want to invert 0 and 1 bits? No.
    1,                          // Number of sendings
    nullptr,                    // No callback to keep/stop sending (if you want to send
                                // SO LONG AS a button is pressed, the function reading the
                                // button state is to be put here).
    17888,                      // initseq
    1432,                       // lo_prefix
    1424,                       // hi_prefix
    0,                          // first_lo_ign
    474,                        // lo_short
    952,                        // lo_long
    0,                          // hi_short (0 => take lo_short)
    0,                          // hi_long  (0 => take lo_long)
    1400,                       // lo_last
    19324,                      // sep
    52                          // nb_bits
  );
```

Thanks! I did see #337 but I wanted to get more insight

**Comment by jgromes** [comment]
First of all, you need to understand the signal sent by the RF433 module - either with SDR, or maybe Google will help. SX1278 can transmit and receive LoRa, FSK and OOK, so if it is anything else, you are most likely out of luck.

Regarding ASK vs OOK, if the module really does expect different amplitudes for different symbols, you probably won't be able to do it with OOK, since you can only do a single amplitude level, or complete 0.

**Reply by Jev1337** [reply]
I see, thanks! I hope to see someone out here with the same case, otherwise I'd have to buy another LoRa.

And according to Google searches, the Generic RF433 Module uses ASK Modulation, however I'm not sure whether the module expect different amplitudes for different symbols.

My mistake was that I assumed that a more expensive module surely had the ability to do what the cheaper module could do.
